parser/impl/parser.cpp

```diff
@@ -7,51 +7,289 @@ namespace parser
 
     // ---- @kureduro
 
+    static lexer::TokenType primitives[] = {
+        lexer::TokenType::IntegerType,
```

Comment:
Why so?

Reply:
Because they're a. **not safe** and b. they **intoxicate the codebase**. One cannot write a c array somewhere and not use it as a c array someday. c arrays also _decay_ to pointers in the template type deductions, so the information about compile-time size is lost. (unlike std::array). The basic rule is **not** to treat C++ as C, and forget/ignore/avoid/ostracise all C API. And if some third party library is in C, then write a C++ wrapper that will hide all these unsafe details.